nixos/tests/prometheus-exporters.nix

```diff
@@ -188,6 +188,42 @@ let
       '';
     };
 
+    mail = {
+      exporterConfig = {
+        enable = true;
+        user = "mailexporter";
+        configuration = {
+          monitoringInterval = "2s";
+          mailCheckTimeout = "10s";
+          servers = [ {
+            name = "testserver";
+            server = "localhost";
+            port = 25;
+            from = "mailexporter@localhost";
+            to = "mailexporter@localhost";
+            detectionDir = "/var/spool/mail/mailexporter/new";
+          } ];
+        };
+      };
+      metricProvider = {
+        services.postfix.enable = true;
+        systemd.services.prometheus-mail-exporter = {
+          after = [ "postfix.service" ];
+          requires = [ "postfix.service" ];
+          preStart = ''
+            cd /var/spool/mail && mkdir -p mailexporter/new
```

Comment:
Does this really have to be there? We should prefer using `RuntimeDirectory` or `StateDirectory` for data bound to a specific systemd service. We can probably also use `DyanmicUser=true` here as the process only requires network access?

Reply:
The exporter needs filesystem access to read the received test mails. This section is just to make sure that the mail directory exists before the exporter starts in the test.

Reply:
Ok, then it is probably fine. Could we restrict it nonetheless? I don't think it needs write access to most of the filesystem etc…

Reply:
I added some fs restrictions, switched to using `WorkingDirectory` instead of `cd` and also rebased on master.